essencium-backend/src/main/java/de/frachtwerk/essencium/backend/service/AbstractUserService.java

```diff
@@ -83,7 +94,22 @@ private void setup() {
   }
 
   @Override
-  public USER loadUserByUsername(final String username) throws UsernameNotFoundException {
+  public UserDetails loadUserByUsername(final String username) throws UsernameNotFoundException {
+    if (username == null || username.isEmpty()) {
+      throw new UsernameNotFoundException("username is empty");
+    }
+    if (username.contains(ApiTokenUser.USER_SPLITTER)) {
+      var split = username.split(ApiTokenUser.USER_SPLITTER);
+      if (split.length == 2) {
+        return apiTokenUserRepository
+            .findById(UUID.fromString(split[1]))
+            .orElseThrow(() -> new UsernameNotFoundException("token not found"));
+      }
```

Comment:
`ApiTokenUser.USER_SPLITTER` refers to `:`.  According to RFC 5322, the local part of mail addresses may only contain letters and numbers as well as certain other characters (`A-Za-z0-9.!#$%&'*+-/=?^_\`{|}~`). The only exception: the local part is surrounded by double superscripts. Then, by definition, the characters `"(),:;<>@[\\]` may also be included. I would say that the colon can be reliably used as a separator, as we can assume that it does not occur in mail addresses. `@Email` and `@ValidEmail` (as we use it in the `UserDto`) do not allow the colon as part of mail addresses.

Reply:
Not sure if you got my point, also the code above doesn't directly show what I mean, it only shows where I would insert the `else`.

I do _not_ question if the colon can be used as separator.

The current situation is, that if you have two colons `A:B:C` then split will give you an array of length 3 and thus it would not qualify as ApiTokenUser because for an ApiTokenUser we only expect one colon `:`.

Reply:
Where should the second (third, ...) colon come from?

Reply:
True, if we only call it internally, fine. Although code might change in the future.